File: 프로그래머스/unrated/159994. 카드 뭉치/카드 뭉치.cpp
#include <string>
#include <vector>

using namespace std;

string solution(vector<string> c1, vector<string> c2, vector<string> goal) {
    string ans = "Yes";
    int i=0,j=0;

    for(int k=0;k<goal.size();k++) {
        if(goal[k] == c1[i]){
            i+=1;
        }else if (goal[k] == c2[j]){
            j+=1;
        }else{
            ans="No";
            break;
        }
    }

    return ans;
}

File: 프로그래머스/unrated/159994. 카드 뭉치/카드 뭉치.java
import java.util.*;


class Solution {

    
        public String solution(String[] cards1, String[] cards2, String[] goal) {
        String answer = "Yes";
        Queue<String> queueForCard1 = new LinkedList<>();
        Queue<String> queueForCard2 = new LinkedList<>();

        for(String card : cards1){
            queueForCard1.add(card);
        }
        for(String card : cards2){
            queueForCard2.add(card);
        }

        for(String word : goal){
            if(word.equals(queueForCard1.peek())){
                queueForCard1.remove();
            }else if (word.equals(queueForCard2.peek())){
                queueForCard2.remove();
            }else{
                answer = "No";
                break;
            }
        }

        return answer;
    }
    
    
}